src/autorestSession.ts

```diff
@@ -0,0 +1,71 @@
+import { Host, Session, startSession } from "@autorest/extension-base";
+import { CodeModel, codeModelSchema } from "@autorest/codemodel";
+import { extractAutorestOptions } from "./utils/autorestOptions";
+import { PackageDetails } from "./models/packageDetails";
+import { TracingInfo } from "./models/clientDetails";
+
+export interface AutorestOptions {
+  restLevelClient?: boolean;
+  azureArm?: boolean;
+  addCredentials?: boolean;
+  credentialKeyHeaderName?: string;
+  credentialScopes?: string[];
+  srcPath: string;
+  packageDetails: PackageDetails;
+  licenseHeader: boolean;
```

Comment:
In the existing code base, it is `shouldGenerateLicense`. The `licenseHeader` makes more sense. Please make sure corresponding change is done in the places where `shouldGenerateLicense` is used.